File: src/main/java/com/aliiensmp/aliienResize/Commands/AbstractResizeCommand.java
package com.aliiensmp.aliienResize.Commands;

import co.aikar.commands.BaseCommand;
import com.aliiensmp.aliienResize.AliienResize;
import com.aliiensmp.aliienResize.Listeners.PlayerConnectionListener;
import org.bukkit.attribute.Attribute;
import org.bukkit.entity.Player;

import java.util.Optional;

public abstract class AbstractResizeCommand extends BaseCommand {

    protected final AliienResize plugin;

    public AbstractResizeCommand(final AliienResize plugin) {
        this.plugin = plugin;
    }

    /**
     * Changes the {@code Attribute.GENERIC_SCALE} of the player safely on Folia.
     *
     * @param target the player whose scale is changing
     * @param scale the new scale value
     * @param onSuccess callback for messages/sounds after the scale is applied
     */
    protected void applyScale(final Player target, final double scale, final Runnable onSuccess) {
        target.getScheduler().run(plugin, scheduledTask -> {

            Optional.ofNullable(target.getAttribute(Attribute.GENERIC_SCALE))
                    .ifPresent(attribute -> attribute.setBaseValue(scale));

            PlayerConnectionListener.cache.put(target.getUniqueId(), scale);

            if (onSuccess != null) {
                onSuccess.run();
            }

        }, null);
    }
}

File: src/main/kotlin/com/aliiensmp/aliienResize/commands/AbstractResizeCommand.kt
package com.aliiensmp.aliienResize.commands

import co.aikar.commands.BaseCommand
import AliienResize
import com.aliiensmp.aliienResize.listeners.PlayerConnectionListener
import org.bukkit.attribute.Attribute
import org.bukkit.entity.Player

abstract class AbstractResizeCommand(protected val plugin: AliienResize) : BaseCommand() {

    protected fun applyScale(target: Player, scale: Double, onSuccess: (() -> Unit)? = null) {
        target.scheduler.run(plugin, { _ ->
            target.getAttribute(Attribute.GENERIC_SCALE)?.baseValue = scale
            PlayerConnectionListener.cache[target.uniqueId] = scale

            onSuccess?.invoke()
        }, null)
    }
}